src/scenes/Review/EditProfile.jsx

```
@@ -0,0 +1,112 @@
+import { updateServiceMember } from 'scenes/ServiceMembers/ducks';
+
+import DutyStationSearchBox from 'scenes/ServiceMembers/DutyStationSearchBox';
+
+import './Review.css';
+import profileImage from './images/profile.png';
+
+const editProfileFormName = 'edit_profile';
+
+let EditProfileForm = props => {
+  const { onCancel, schema, handleSubmit, pristine, submitting, valid } = props;
+  return (
+    <form onSubmit={handleSubmit}>
+      <img src={profileImage} alt="" /> Profile
+      <hr />
+      <h3 className="sm-heading">Edit Profile:</h3>
+      <SwaggerField fieldName="first_name" swagger={schema} required />
+      <SwaggerField fieldName="middle_name" swagger={schema} />
+      <SwaggerField fieldName="last_name" swagger={schema} required />
+      <SwaggerField fieldName="suffix" swagger={schema} />
+      <hr className="spacer" />
+      <SwaggerField fieldName="affiliation" swagger={schema} required />
+      <SwaggerField fieldName="rank" swagger={schema} required />
+      <SwaggerField fieldName="edipi" swagger={schema} required />
+      <Field name="current_station" component={DutyStationSearchBox} />
+      <button type="submit" disabled={pristine || submitting || !valid}>
+        Save
+      </button>
+      <button type="button" disabled={submitting} onClick={onCancel}>
+        Cancel
+      </button>
+    </form>
+  );
+};
+const validateProfileForm = validateAdditionalFields(['current_station']);
+EditProfileForm = reduxForm({
+  form: editProfileFormName,
+  validate: validateProfileForm,
+})(EditProfileForm);
+
+class EditProfile extends Component {
+  returnToReview = () => {
+    const reviewAddress = `/moves/${this.props.match.params.moveId}/review`;
+    this.props.push(reviewAddress);
+  };
+
+  updateProfile = (fieldValues, something, elses) => {
+    fieldValues.current_station_id = fieldValues.current_station.id;
+
+    this.props.updateServiceMember(fieldValues).then(() => {
+      // This promise resolves regardless of error.
```

This comment is not true. If the update fails, the promise is rejected.
When update fails, can we scroll to the top? At the moment, an alert appears there, but you can't see it because it is off screen.
If an update fails (say the server goes down) and then the next attempt is successful, the panel does not close.

---

It does appear to be true. catch() swallows an error (but you can re-throw one) so for now this is necessary. We are tracking error state in redux so I'm ok with us using the promise as simply a way to track the completion of the request. The other option is to have the .catch re-throw. in which case it should also probably be returning the object/error.